[Bug] 测试夹具硬编码日期问题再次导致 CI 持续失败

## 版本确认 / Version Check（必填 / Required）
- [x] 我已同步最新代码 / I am on the latest commit (Fork users: Sync fork first, then re-run Actions)
- 代码版本 / Commit hash:
  - 本地 / Local: `git rev-parse --short HEAD` → 53c8ff46
  - GitHub Actions: backend-tests 工作流日志顶部展示的 commit hash → 53c8ff46（PR #2197）

## 问题描述 / Problem Description

CI（backend-tests / backend-gate）持续失败：`tests/test_pipeline_single_stock_notify.py` 有 2 个测试失败，均报 `AssertionError: 'report_20260815_600519.md' != 'report_20260814_600519.md'`。
**该失败与日期强相关**——测试断言硬编码了文件名日期 `20260814`，而生产代码 `src/core/pipeline.py` 的单股报告文件名按**运行当天**动态生成（`report_{YYYYMMDD}_{code}.md`），系统日期越过 2026-08-14 后实际文件名变为 `report_20260815_600519.md`，断言必失败。
该测试改动为 upstream 已合并 PR #2201/#2211（commit `29421088`）的原样内容，因分支 rebase upstream/main 被带入 PR diff；fork 的 origin/main 落后（base 停在 `3b98aa1d`）导致该上游改动被计入本 PR 的变更范围

失败用例（2 处）：
1. `tests/test_pipeline_single_stock_notify.py::TestPipelineSingleStockNotify::test_process_single_stock_direct_path_keeps_notify_compatibility`
2. `tests/test_pipeline_single_stock_notify.py::TestPipelineSingleStockNotify::test_process_single_stock_saves_report_even_when_notifier_is_unavailable`

这是**测试数据的时间依赖缺陷**，不是生产代码 bug：报告文件名按当天日期生成是正确行为（与既有 `NotificationService` 的 `report_{YYYYMMDD}.md` 命名约定一致）。

## 复现步骤 / Reproduction Steps

1. 运行命令 / Run command `python -m pytest tests/test_pipeline_single_stock_notify.py -q`
2. 系统日期 / System date 在 2026-08-15 及之后（本地已于 2026-08-15 复现；GitHub Actions 上 backend-tests shard 2 必现）
3. 查看 / View 输出：两个用例依次失败，均停在 `self.assertEqual(save_call.kwargs["filename"], "report_20260814_600519.md")` 断言
4. 出现错误 / Error occurs：`AssertionError: 'report_20260815_600519.md' != 'report_20260814_600519.md'`

时间线：测试写于 2026-08-14（当日 `datetime.now()` 得到 20260814，与硬编码一致，通过）；2026-08-15 起实际文件名日期与硬编码不符，必现失败。

## 期望行为 / Expected Behavior

- `save_report_to_file` 收到文件名与生产代码生成逻辑一致：`report_{当天日期}_{代码}.md`
- 测试在任何日期运行时都稳定通过（测试意图是验证单股报告内容与文件名正确传入保存回调，并保持既有的通知发送行为）

## 实际行为 / Actual Behavior

- 2026-08-15 起实际文件名日期为 20260815（`datetime.now().strftime('%Y%m%d')`），断言期望 20260814 → 失败
- 两个用例都在第一个失败断言处停止
- 结果：CI 的 backend-tests shard 2 失败（shard 1 / 3 通过），backend-gate 报 "Backend shards did not all pass: failure"，PR 无法合入

## 错误日志 / Error Logs

参考 PR #2197

## 环境信息 / Environment

- 操作系统 / OS: Windows 11（本地复现）/ GitHub Actions ubuntu-latest（CI）
- Python 版本 / Python version: 3.13.3（本地）/ 3.11（CI），pytest 9.1.1
- 运行方式 / Run mode: GitHub Actions（backend-tests shard 2）+ 本地 pytest
- 相关配置 / Relevant config: 无（测试为离线单测，mock notifier，不依赖网络与数据源）

## 其他信息 / Additional Context

### 根因分析

关键代码：`src/core/pipeline.py`（`_send_single_stock_notification`）

```python
save_report = getattr(self.notifier, "save_report_to_file", None)
if callable(save_report):
    try:
        date_str = datetime.now().strftime('%Y%m%d')   # 文件名日期随运行当天动态生成
        filename = f"report_{date_str}_{stock_code}.md"
        filepath = save_report(report_content, filename=filename)
```

测试夹具：`tests/test_pipeline_single_stock_notify.py`（161 行 / 189 行 / 52 行）写死文件名日期 `20260814`：

```python
self.assertEqual(save_call.kwargs["filename"], "report_20260814_600519.md")
```

2026-08-15 实测：`datetime.now().strftime('%Y%m%d')` → `20260815`，实际文件名 `report_20260815_600519.md`，与硬编码的 `20260814` 不符 → 断言失败。

生产代码按当天日期生成报告文件名是**正确行为**（与 `NotificationService` 的 `report_{YYYYMMDD}.md` 命名约定一致），无需改动；问题完全在测试硬编码日期、与系统时钟隐式耦合。

### 修复方案（已实施并验证）

**修复思路**：只改测试、不改生产代码。测试断言与 mock 中的硬编码日期全部改为按当天动态生成，与 `pipeline.py` / `NotificationService` 的动态命名逻辑保持一致。生产代码 `src/core/pipeline.py` 保持 `datetime.now().strftime('%Y%m%d')` 不动（这是正确语义：每天一份带日期的报告）。

**改动文件**：`tests/test_pipeline_single_stock_notify.py`（+7 / -3，仅测试）

**改动点（4 处）**：

1. 新增导入：
```python
from datetime import datetime
```

2. `_TrackingNotifier._save_report_to_file` 的 fallback 返回值（原 52 行）：
```python
# 修改前
return f"/tmp/{filename or 'report_20260814.md'}"
# 修改后
today = datetime.now().strftime('%Y%m%d')
return f"/tmp/{filename or f'report_{today}.md'}"
```

3. `test_process_single_stock_direct_path_keeps_notify_compatibility` 断言（原 161 行）：
```python
# 修改前
self.assertEqual(save_call.kwargs["filename"], "report_20260814_600519.md")
# 修改后
today = datetime.now().strftime('%Y%m%d')
self.assertEqual(save_call.kwargs["filename"], f"report_{today}_600519.md")
```

4. `test_process_single_stock_saves_report_even_when_notifier_is_unavailable` 断言（原 189 行）：
```python
# 修改前
self.assertEqual(save_call.kwargs["filename"], "report_20260814_600519.md")
# 修改后
today = datetime.now().strftime('%Y%m%d')
self.assertEqual(save_call.kwargs["filename"], f"report_{today}_600519.md")
```

**完整 diff**：

```diff
--- a/tests/test_pipeline_single_stock_notify.py
+++ b/tests/test_pipeline_single_stock_notify.py
@@ -9,6 +9,7 @@ import sys
 import threading
 import time
 import unittest
+from datetime import datetime
 from pathlib import Path
 from types import SimpleNamespace
 from unittest.mock import MagicMock, patch
@@ -49,7 +50,8 @@ class _TrackingNotifier:

     def _save_report_to_file(self, content, filename=None):
         self.saved_reports.append((content, filename))
-        return f"/tmp/{filename or 'report_20260814.md'}"
+        today = datetime.now().strftime('%Y%m%d')
+        return f"/tmp/{filename or f'report_{today}.md'}"

     def _send(
         self,
@@ -158,7 +160,8 @@ class TestPipelineSingleStockNotify(unittest.TestCase):
         pipeline.notifier.generate_brief_report.assert_called_once_with([result])
         save_call = pipeline.notifier.save_report_to_file.call_args
         self.assertEqual(save_call.args[0], "brief:600519")
-        self.assertEqual(save_call.kwargs["filename"], "report_20260814_600519.md")
+        today = datetime.now().strftime('%Y%m%d')
+        self.assertEqual(save_call.kwargs["filename"], f"report_{today}_600519.md")
         pipeline.notifier.send.assert_called_once_with(
             "brief:600519",
             email_stock_codes=["600519"],
@@ -186,7 +189,8 @@ class TestPipelineSingleStockNotify(unittest.TestCase):
         self.assertIsNotNone(result)
         save_call = pipeline.notifier.save_report_to_file.call_args
         self.assertEqual(save_call.args[0], "single:600519")
-        self.assertEqual(save_call.kwargs["filename"], "report_20260814_600519.md")
+        today = datetime.now().strftime('%Y%m%d')
+        self.assertEqual(save_call.kwargs["filename"], f"report_{today}_600519.md")
         pipeline.notifier.send.assert_not_called()
```

**验证结果**：

```bash
python -m pytest tests/test_pipeline_single_stock_notify.py -q
# 修复前：2 failed, 7 passed（8.76s）
# 修复后：9 passed（4.33s）

python -m pytest tests/test_pipeline_single_stock_notify.py tests/test_web_intent_resolver.py tests/test_name_to_code_resolver.py -q
# 127 passed —— 意图层 / 名称解析相关测试不受影响
```

**注意事项**：

- 请勿通过修改 `src/core/pipeline.py` 把文件名日期改为固定值来迁就测试——那会破坏"每天一份报告"的命名语义，也与 `NotificationService`（`src/notification.py:2535-2537`，`report_{YYYYMMDD}.md`）的既有约定不一致。
- 该修复仅涉及测试文件，生产行为零改动。

### 同类隐患

本项目已多次出现"测试夹具硬编码日期、隔天 CI 失败"的同类问题（upstream 曾以 `c7ca990b` "make dividend TTM dates date-relative to fix CI (#2204) (#2206)" 修复 yfinance 分红日期的同类问题）。建议全库排查测试中硬编码的日历日期（`20260814`、`2026-08-14` 等），统一改为相对当前日期生成或 `freezegun` 冻结。

### 影响面

- CI `backend-tests (2/3)` / `backend-gate` 持续失败，阻断 PR 合入

--- a/tests/test_pipeline_single_stock_notify.py
+++ b/tests/test_pipeline_single_stock_notify.py
@@ -9,6 +9,7 @@ import sys
 import threading
 import time
 import unittest
+from datetime import datetime
 from pathlib import Path
 from types import SimpleNamespace
 from unittest.mock import MagicMock, patch
@@ -49,7 +50,7 @@ class _TrackingNotifier:
 
     def _save_report_to_file(self, content, filename=None):
         self.saved_reports.append((content, filename))
-        return f"/tmp/{filename or 'report_20260814.md'}"
+        return f"/tmp/{filename or 'report.md'}"
 
     def _send(
         self,
@@ -89,6 +90,8 @@ def _make_result(code: str, success: bool = True) -> AnalysisResult:
 
 
 class TestPipelineSingleStockNotify(unittest.TestCase):
+    _FROZEN_REPORT_TIME = datetime(2030, 1, 2, 12, 0, 0)
+
     @staticmethod
     def _build_batch_pipeline() -> StockAnalysisPipeline:
         pipeline = StockAnalysisPipeline.__new__(StockAnalysisPipeline)
@@ -146,19 +149,21 @@ class TestPipelineSingleStockNotify(unittest.TestCase):
         pipeline.analyze_stock = MagicMock(return_value=_make_result("600519"))
         pipeline.notifier = _TrackingNotifier()
 
-        result = pipeline.process_single_stock(
-            code="600519",
-            skip_analysis=False,
-            single_stock_notify=True,
-            report_type=ReportType.BRIEF,
-            analysis_query_id="query-1",
-        )
+        with patch("src.core.pipeline.datetime") as mock_datetime:
+            mock_datetime.now.return_value = self._FROZEN_REPORT_TIME
+            result = pipeline.process_single_stock(
+                code="600519",
+                skip_analysis=False,
+                single_stock_notify=True,
+                report_type=ReportType.BRIEF,
+                analysis_query_id="query-1",
+            )
 
         self.assertIsNotNone(result)
         pipeline.notifier.generate_brief_report.assert_called_once_with([result])
         save_call = pipeline.notifier.save_report_to_file.call_args
         self.assertEqual(save_call.args[0], "brief:600519")
-        self.assertEqual(save_call.kwargs["filename"], "report_20260814_600519.md")
+        self.assertEqual(save_call.kwargs["filename"], "report_20300102_600519.md")
         pipeline.notifier.send.assert_called_once_with(
             "brief:600519",
             email_stock_codes=["600519"],
@@ -175,18 +180,20 @@ class TestPipelineSingleStockNotify(unittest.TestCase):
         pipeline.notifier = _TrackingNotifier()
         pipeline.notifier.is_available.return_value = False
 
-        result = pipeline.process_single_stock(
-            code="600519",
-            skip_analysis=False,
-            single_stock_notify=True,
-            report_type=ReportType.SIMPLE,
-            analysis_query_id="query-1",
-        )
+        with patch("src.core.pipeline.datetime") as mock_datetime:
+            mock_datetime.now.return_value = self._FROZEN_REPORT_TIME
+            result = pipeline.process_single_stock(
+                code="600519",
+                skip_analysis=False,
+                single_stock_notify=True,
+                report_type=ReportType.SIMPLE,
+                analysis_query_id="query-1",
+            )
 
         self.assertIsNotNone(result)
         save_call = pipeline.notifier.save_report_to_file.call_args
         self.assertEqual(save_call.args[0], "single:600519")
-        self.assertEqual(save_call.kwargs["filename"], "report_20260814_600519.md")
+        self.assertEqual(save_call.kwargs["filename"], "report_20300102_600519.md")
         pipeline.notifier.send.assert_not_called()
 
     def test_process_single_stock_updates_saved_diagnostics_after_notification(self):

